modules/csvparser/csvparser.c

```diff
@@ -165,6 +165,89 @@ dispatch_key_formatter(gchar *prefix)
   return prefix ? _format_key_for_prefix : _return_key;
 }
 
+gboolean
+_should_drop_message(CSVParser *self)
+{
+  return (self->template_options.on_error & ON_ERROR_DROP_MESSAGE);
+}
+
+gboolean
+csv_parser_get_drop_invalid(LogParser *s)
+{
+  CSVParser *self = (CSVParser *) s;
+  return _should_drop_message(self);
+
+}
+
+static gboolean
+_process_column_on_invalid_type_cast(CSVParser *self, gchar *current_column_name,
+                                     LogMessageValueType *current_type, GError *error)
+{
+  if(self->template_options.on_error & ON_ERROR_FALLBACK_TO_STRING)
+    {
+      if (!(self->template_options.on_error & ON_ERROR_SILENT))
+        {
+          msg_debug("csv-parser: error casting value to the type specified, trying string type because on-error(\"fallback-to-string\") was specified",
+                    evt_tag_str("error", error->message));
+          g_clear_error(&error);
+        }
+      *current_type = LM_VT_STRING;
+      return TRUE;
+    }
+  else if (self->template_options.on_error & ON_ERROR_DROP_PROPERTY)
+    {
+      if (!(self->template_options.on_error & ON_ERROR_SILENT))
+        {
+          msg_debug("csv-parser: error casting value to the type specified, dropping property because on-error(\"drop-property\") was specified",
+                    evt_tag_str("error", error->message));
+          g_clear_error(&error);
+        }
+    }
+  return FALSE;
+}
+
+
+static gboolean
+_process_column(CSVParser *self, CSVScanner *scanner, LogMessage *msg, GList *column_l, GString *key_scratch,
+                key_formatter_t _key_formatter)
+{
+  CSVParserColumn *current_column = column_l->data;
+  LogMessageValueType current_column_type = current_column->type;
+  const gchar *current_value = csv_scanner_get_current_value(scanner);
+
+  GError *error = NULL;
+  gboolean should_set_value = TRUE;
+
+  if (!type_cast_validate(current_value, current_column_type, &error))
+    {
+      gboolean should_drop = type_cast_drop_helper(self->template_options.on_error, current_value,
+                                                   log_msg_value_type_to_str(current_column_type));
+      if (should_drop)
+        {
+          if (!(self->template_options.on_error & ON_ERROR_SILENT))
+            {
+              msg_debug("csv-parser: error casting value to the type specified",
+                        evt_tag_str("error", error->message)
+                       );
```

Comment:
`drop_helper()` already logs error, I think this is not needed.

Reply:
I left it here because `drop_helper` does not print out the exact `error` and it might help with locating the exact error.